GPU/Workflow/src/GPUWorkflowSpec.cxx

```diff
@@ -656,6 +732,39 @@ DataProcessorSpec getGPURecoWorkflowSpec(gpuworkflow::CompletionPolicyData* poli
       }
 
       const auto& holdData = TPCTrackingDigitsPreCheck::runPrecheck(&ptrs, processAttributes->config.get());
+
+      // update the calibration objects in case they changed in the CCDB
+      const CalibdEdxContainer* dEdxCalibContainer = processAttributes->dEdxCalibContainer.get();
+
+      if (dEdxCalibContainer->isCorrectionCCDB(CalibsdEdx::CalThresholdMap)) {
+        pc.inputs().get<std::unordered_map<std::string, o2::tpc::CalDet<float>>*>("tpcthreshold");
+      }
+
+      if (dEdxCalibContainer->isCorrectionCCDB(CalibsdEdx::CalResidualGainMap)) {
+        pc.inputs().get<std::unordered_map<std::string, o2::tpc::CalDet<float>>*>("tpcgainresidual");
+      }
+
+      if (dEdxCalibContainer->isCorrectionCCDB(CalibsdEdx::CalGainMap)) {
+        pc.inputs().get<o2::tpc::CalDet<float>*>("tpcgain");
+      }
+
+      if (dEdxCalibContainer->isCorrectionCCDB(CalibsdEdx::CalTopologyPol)) {
+        pc.inputs().get<o2::tpc::CalibdEdxTrackTopologyPolContainer*>("tpctopologygain");
+      }
+
+      if (dEdxCalibContainer->isCorrectionCCDB(CalibsdEdx::CalTimeGain)) {
+        pc.inputs().get<o2::tpc::CalibdEdxCorrection*>("tpctimegain");
+      }
+
+      if (processAttributes->updateCalibs) {
```

Comment:
Do I understand correctly that we take this path only when the calib object has actually changed? We will not update with the same object over and over?

Reply:
Yes this function triggers a check if the CCDB object has changed and if it has the `finaliseCCDB` function will be called.